File: rentacar-frontend/src/services/userService.ts
import axiosInstance from './axiosInstance';
import { type GetUserProfileResponse, type UpdateUserPasswordRequest } from '../models/types';

const UserService = {
    getProfile: async (): Promise<GetUserProfileResponse> => {
        const response = await axiosInstance.get<GetUserProfileResponse>('/users/me');
        return response.data;
    },

    updatePassword: async (request: UpdateUserPasswordRequest): Promise<void> => {
        await axiosInstance.put('/users/me/password', request);
    }
};

export default UserService;

File: rentACar/src/main/java/com/sinan/rentACar/business/abstracts/UserService.java
package com.sinan.rentACar.business.abstracts;

import com.sinan.rentACar.business.requests.UpdateUserPasswordRequest;
import com.sinan.rentACar.business.responses.GetUserProfileResponse;

public interface UserService {
    GetUserProfileResponse getProfile(String userEmail);
    void updatePassword(String userEmail, UpdateUserPasswordRequest request);
}
